automation_tools/__init__.py

```diff
@@ -1756,8 +1756,10 @@ def copy_ssh_key(from_host, to_host):
     :param to_host: A string. Hostname on to which the ssh-key will be copied.
 
     """
-    execute(lambda: run('[ ! -f ~/.ssh/id_rsa.pub ] && ssh-keygen '
-                        '-f ~/.ssh/id_rsa -t rsa -N \'\''), host=from_host)
+    execute(lambda: run(
+        '[ ! -f ~/.ssh/id_rsa.pub ] && '
+        'ssh-keygen -f ~/.ssh/id_rsa -t rsa -N \'\'', warn_only=True),
```

Comment:
Talked with @jyejare and the warn_only is needed to ensure that fabric will not stop the execution if the private key since the checking will return false (non-zero). Anyway there is another check later to ensure the private key exists.